controllers/metal3remediation_controller.go

```diff
@@ -141,76 +149,96 @@ func (r *Metal3RemediationReconciler) reconcileNormal(ctx context.Context,
 	}
 
 	if remediationType == infrav1.RebootRemediationStrategy {
-		// If no phase set, default to running
+		// If no phase set, default to running and set time and retry count
 		if remediationMgr.GetRemediationPhase() == "" {
 			remediationMgr.SetRemediationPhase(infrav1.PhaseRunning)
+			now := metav1.Now()
+			remediationMgr.SetLastRemediationTime(&now)
+			remediationMgr.IncreaseRetryCount()
```

Comment:
This looks a little odd, given that we haven't performed a remediation yet but we're setting a "last remediation time" and a retry count of 1. I guess we're reusing existing fields to have slightly different meanings. The latter at least seems unnecessary.

What is the impact on upgrades from previous versions of the controller (i.e. when there are remediations in flight)?

Reply:
removed `IncreaseRetryCount`